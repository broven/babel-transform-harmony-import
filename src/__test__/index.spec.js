const babel = require('@babel/core');
const plugin = require('../index');


const opt = {
    babelrc: false,
    plugins: [
      [plugin, {
          localName: ['testModule', 'testModule1', 'importModule2', 'Nerv']
      }]
    ]
  };


  var example = `
import {testModule} from 'origin';
import testModule1 from 'origin';
import * as importModule2 from 'origin';
import Nerv from 'nervjs';
import dsa from 'dsa';
`;

it('simple Usage', () => {
    const {code} = babel.transformSync(example, opt);
    expect(code).toMatchSnapshot();
  });