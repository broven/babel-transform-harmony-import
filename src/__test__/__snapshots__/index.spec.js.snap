// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple Usage 1`] = `
"function __babel_transform_harmony_import__() {}

import { testModule } from 'origin';

__babel_transform_harmony_import__(testModule);

import testModule1 from 'origin';

__babel_transform_harmony_import__(testModule1);

import * as importModule2 from 'origin';

__babel_transform_harmony_import__(importModule2);

import Nerv from 'nervjs';

__babel_transform_harmony_import__(Nerv);

import dsa from 'dsa';"
`;
